src/internal/converters/ics/ics_test.go

```diff
@@ -1079,3 +1079,152 @@ func (suite *ICSUnitSuite) TestAttachments() {
 		})
 	}
 }
+
+func (suite *ICSUnitSuite) TestCancellations() {
+	table := []struct {
+		name         string
+		cancelledIds []string
+		expected     string
+	}{
+		{
+			name: "single",
+			cancelledIds: []string{
+				"OID.DEADBEEF=.2024-01-25",
+			},
+			expected: "EXDATE:20240125",
+		},
+		{
+			name: "multiple",
+			cancelledIds: []string{
+				"OID.DEADBEEF=.2024-01-25",
+				"OID.LIVEBEEF=.2024-02-26",
+			},
+			expected: "EXDATE:20240125,20240226",
+		},
+	}
+
+	for _, tt := range table {
+		suite.Run(tt.name, func() {
+			t := suite.T()
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			// convert event to bytes
+			writer := kjson.NewJsonSerializationWriter()
+			defer writer.Close()
+
+			e := baseEvent()
+
+			e.SetIsCancelled(ptr.To(true))
+			e.SetAdditionalData(map[string]any{
+				"cancelledOccurrences": tt.cancelledIds,
+			})
+
+			err := writer.WriteObjectValue("", e)
+			require.NoError(t, err, "serializing contact")
+
+			bts, err := writer.GetSerializedContent()
+			require.NoError(t, err, "getting serialized content")
+
+			out, err := FromJSON(ctx, bts)
+			require.NoError(t, err, "converting to ics")
+
+			assert.Contains(t, out, tt.expected, "cancellation exrule")
+		})
+	}
+}
+
+func (suite *ICSUnitSuite) TestEventExceptions() {
+	table := []struct {
+		name  string
+		event func() *models.Event
+		check func(string)
+	}{
+		{
+			name: "single exception",
```

Comment:
could you please also add a test with > 1 exception? That will act as a regression test to make sure we don't accidentally lose the ability to handle multiple exceptions in the future